rust/lancedb/src/table.rs

```diff
@@ -1649,345 +1647,211 @@ impl NativeTable {
-                opts.replace,
-            )
-            .await?;
-        Ok(())
+    // Helper to get num_sub_vectors with default calculation
+    fn get_num_sub_vectors(provided: Option<u32>, dim: u32) -> u32 {
+        provided.unwrap_or_else(|| suggested_num_sub_vectors(dim))
     }
 
-    async fn create_label_list_index(&self, field: &Field, opts: IndexBuilder) -> Result<()> {
-        if !supported_label_list_data_type(field.data_type()) {
-            return Err(Error::Schema {
-                message: format!(
-                    "A LabelList index cannot be created on the field `{}` which has data type {}",
-                    field.name(),
-                    field.data_type()
-                ),
-            });
+    // Helper to extract vector dimension from field
+    fn get_vector_dimension(field: &Field) -> Result<u32> {
+        match field.data_type() {
+            arrow_schema::DataType::FixedSizeList(_, n) => Ok(*n as u32),
+            _ => Ok(infer_vector_dim(field.data_type())? as u32),
         }
-
-        let mut dataset = self.dataset.get_mut().await?;
-        let lance_idx_params = lance_index::scalar::ScalarIndexParams {
-            force_index_type: Some(lance_index::scalar::ScalarIndexType::LabelList),
-        };
-        dataset
-            .create_index(
-                &[field.name()],
-                IndexType::LabelList,
-                None,
-                &lance_idx_params,
-                opts.replace,
-            )
-            .await?;
-        Ok(())
     }
 
-    async fn create_fts_index(
+    // Convert LanceDB Index to Lance IndexParams
+    async fn make_index_params(
         &self,
         field: &Field,
-        fts_opts: FtsIndexBuilder,
-        replace: bool,
-    ) -> Result<()> {
-        if !supported_fts_data_type(field.data_type()) {
-            return Err(Error::Schema {
-                message: format!(
-                    "A FTS index cannot be created on the field `{}` which has data type {}",
-                    field.name(),
-                    field.data_type()
-                ),
-            });
+        index_opts: Index,
+    ) -> Result<Box<dyn lance::index::IndexParams>> {
+        match index_opts {
+            Index::Auto => {
+                if supported_vector_data_type(field.data_type()) {
+                    // Use IvfPq as the default for auto vector indices
+                    let dim = Self::get_vector_dimension(field)?;
+                    let num_partitions = self.get_num_partitions(None, false, None).await?;
+                    let num_sub_vectors = Self::get_num_sub_vectors(None, dim);
+                    let lance_idx_params = lance::index::vector::VectorIndexParams::ivf_pq(
+                        num_partitions as usize,
+                        /*num_bits=*/ 8,
+                        num_sub_vectors as usize,
+                        lance_linalg::distance::MetricType::L2,
+                        50, // max_iterations
```

Comment:
Super minor nit but you're mixing conventions here.
```suggestion
                        /*num_bits=*/ 8,
                        num_sub_vectors as usize,
                        lance_linalg::distance::MetricType::L2,
                        /*max_iterations=*/ 50,
```